app/controllers/api_controller/collection_config.rb

```diff
@@ -0,0 +1,110 @@
+class ApiController
+  class CollectionConfig < Config::Options
+    def collection?(collection_name)
```

Comment:
I'm noticing most of these methods take `collection_name` as an argument. I'm wondering if this is calling for a further abstraction - Collection objects? Probably not as part of this PR though, that would make it a much larger refactoring